src/utilities.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright libCellML Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "utilities.h"
+
+#include <stdexcept>
+
+namespace libcellml {
+
+bool canConvertToDouble(const std::string &candidate)
+{
+    bool canConvert = false;
+    // Try to convert the input string to double.
+    try
+    {
+        std::stod(candidate);
+        canConvert = true;
+    } catch (std::invalid_argument) {
+        canConvert = false;
+    } catch (std::out_of_range) {
+        canConvert = false;
+    }
+
+    return canConvert;
+}
```

Comment:
> Agreed in principle, it's just that here both types of exception result in the same outcome, i.e. canConvert = false;, so I am not sure what the point of being explicit is here.

Being consistently explicit can force us to be aware of and check for less-likely error modes that might be overlooked with a `catch (...)`. It might also be useful to someone interactively debugging to see what kind of error was caught.